server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -603,6 +603,15 @@ abstract class ResultStrategy<
         TB extends InternalMultiBucketAggregation.InternalBucket> implements Releasable {
 
         private InternalAggregation[] buildAggregations(long[] owningBucketOrds) throws IOException {
+            int requiredSizeLocal;
+            long minDocCountLocal;
+            if (context.isConcurrentSegmentSearchEnabled()) {
+                requiredSizeLocal = Integer.MAX_VALUE;
```

Comment:
```suggestion
                requiredSizeLocal = ArrayUtil.MAX_ARRAY_LENGTH - 1;
```

This is the max we could (the org.apache.lucene.util.PriorityQueue limit), it is very close to Integer.MAX_VALUE